src/main/java/com/botdetector/BotDetectorPlugin.java

```diff
@@ -657,11 +659,41 @@ private void onMenuEntryAdded(MenuEntryAdded event)
 		}
 	}
 
+	@Subscribe
+	private void onMenuOpened(MenuOpened event)
+	{
+		if (config.highlightPredictOption() != NOT_REPORTED)
+		{
+			return;
+		}
+
+		// Do this once when the menu opens
+		// Avoids having to loop the menu entries on every 'added' event
+		// Although, flashes red for one client tick
+
+		MenuEntry[] menuEntries = event.getMenuEntries();
+		for (MenuEntry entry : menuEntries)
+		{
+			int type = entry.getType();
+			if (type >= MenuAction.MENU_ACTION_DEPRIORITIZE_OFFSET)
+			{
+				type -= MenuAction.MENU_ACTION_DEPRIORITIZE_OFFSET;
+			}
+
+			if (type == MenuAction.RUNELITE_PLAYER.getId()
+				&& entry.getOption().equals(HIGHLIGHTED_PREDICT_OPTION))
+			{
+				entry.setOption(getPredictOption(client.getCachedPlayers()[entry.getIdentifier()].getName()));
```

Comment:
```suggestion
				Player player = client.getCachedPlayers()[entry.getIdentifier()];
				if (player != null)
				{
					entry.setOption(getPredictOption(player.getName()));
				}
```
I just though we probably should have a null check here just in case.